setup.py

```diff
@@ -165,466 +26,206 @@ def _get_interop_filename():
-                    data_files[1][j] = filename.format(build_lib=build_lib)
-                dest = data_files[0].format(install_platlib=install_platlib)
-                self.data_files[i] = dest, data_files[1]
+                self.announce("Writing configured.props...", level=distutils.log.INFO)
+                _write_configure_props()
 
-        return install_data.install_data.run(self)
+        for lib in dotnet_modules:
+            output = os.path.join(os.path.abspath(self.build_lib), lib.args.pop("output"))
+            rename = lib.args.pop("rename", {})
 
+            opts = sum(
+                [
+                    ["--" + name.replace("_", "-"), value]
+                    for name, value in lib.args.items()
+                ],
+                [],
+            )
 
-class InstallPythonnet(install.install):
-    user_options = install.install.user_options + [("xplat", None, None)]
+            opts.extend(["--configuration", self.dotnet_config])
+            opts.extend(["--output", output])
 
-    def initialize_options(self):
-        install.install.initialize_options(self)
-        self.xplat = None
+            self.announce("Running dotnet build...", level=distutils.log.INFO)
+            self.spawn(["dotnet", "build", lib.path] + opts)
 
-    def finalize_options(self):
-        install.install.finalize_options(self)
+            for k, v in rename.items():
+                source = os.path.join(output, k)
+                dest = os.path.join(output, v)
 
-    def run(self):
-        if self.xplat:
-            _update_xlat_devtools()
-        return install.install.run(self)
+                if os.path.isfile(source):
+                    try:
+                        os.remove(dest)
+                    except OSError:
+                        pass
 
-if bdist_wheel:
-    class BDistWheelPythonnet(bdist_wheel.bdist_wheel):
-        user_options = bdist_wheel.bdist_wheel.user_options + [("xplat", None, None)]
+                    self.move_file(src=source, dst=dest, level=distutils.log.INFO)
+                else:
+                    self.warn("Can't find file to rename: {}, current dir: {}".format(source, os.getcwd()))
 
-        def initialize_options(self):
-            bdist_wheel.bdist_wheel.initialize_options(self)
-            self.xplat = None
+        if other_modules:
+            self.distribution.ext_modules = other_modules
+            super().run()
+            self.distribution.ext_modules = orig_modules
+        # If no modules need to be compiled, skip
 
-        def finalize_options(self):
-            bdist_wheel.bdist_wheel.finalize_options(self)
 
-        def run(self):
-            if self.xplat:
-                _update_xlat_devtools()
-            return bdist_wheel.bdist_wheel.run(self)
+class bdist_wheel_patched(bdist_wheel):
+    def finalize_options(self):
+        # Monkey patch bdist_wheel to think the package is pure even though we
+        # include DLLs
```

Comment:
What is the purpose of this? "pure" in this context means arch-independent? Better state so in the comment then.

Reply:
I have removed this for now, I copied it over from the `clr-loader` `setup.py`. We will need this once our build is platform independent, as `setuptools` tries to be smart and always appends the full platform abi by default if DLLs show up (or if `build_ext` is used, could be that we just need to add a new command instead of abusing this one).